### PR TITLE
Add Unit Tests

### PR DESCRIPTION
**Pull request by Tastebuder**
Screenshot 2024-12-05 at 19 19 50

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 plugins {
-//    id 'java'
```
Comment (CameronMukherjee) [inline build.gradle:2]: Is there any reason we disabled this?

File: src/main/java/com/hexploits/oss/sdk/presidio/anonymizer/strategy/MaskAnonymizerStrategy.java [hunk 2]
```diff
@@ -18,6 +18,9 @@ public class MaskAnonymizerStrategy implements AnonymizerStrategy {
     private final boolean fromEnd;
 
     public MaskAnonymizerStrategy(String maskingChar, int charsToMask, boolean fromEnd) {
+        if (charsToMask < 0) {
+            throw new IllegalArgumentException("charsToMask cannot be negative");
+        }
```
Comment (CameronMukherjee) [inline src/main/java/com/hexploits/oss/sdk/presidio/anonymizer/strategy/MaskAnonymizerStrategy.java:23]: Can we add a new line?

File: src/test/java/com/hexploits/oss/sdk/presidio/analyzer/PresidioAnalyzerClientTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    @Test
+    void testAnalyze() throws IOException {
+        try (MockServerClient client = new MockServerClient("localhost", 8080)) {
+            client.when(HttpRequest.request().withMethod("POST").withPath("/analyze"))
+              .respond(HttpResponse.response()
+                .withStatusCode(200)
+                .withHeader("Content-Type", "application/json")
+                .withBody(
+                  "[{\"analysis_explanation\":null,\"recognition_metadata\":{\"recognizer_identifier\":\"recognizerIdentifier\",\"recognizer_name\":\"recognizerName\"},\"entity_type\":\"field2\",\"score\":0.5,\"start\":1,\"end\":2}]"));
```
Comment (CameronMukherjee) [inline src/test/java/com/hexploits/oss/sdk/presidio/analyzer/PresidioAnalyzerClientTest.java:42]: Any chance we can move these to the test resource folder and read them in for the tests?

Might be more re-usable that way.

**Review by CameronMukherjee — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (Tastebuder) [reply build.gradle:2]: This was just commented out, so got rid since it was commented out

File: src/test/resources/AnalyzeResponse.json
Comment (CameronMukherjee) [inline src/test/resources/AnalyzeResponse.json:1]: Can we name this something that helps us re-use it across tests?

Comment (CameronMukherjee) [reply src/test/resources/AnalyzeResponse.json:1]: Ideally something like: `analyse_reponse_(the type of response)`.

Would be great to also get this in it's own folder in `test`?

Comment (CameronMukherjee) [reply src/test/resources/AnalyzeResponse.json:1]: Final request related to this, can we make it all lower case? Filename seems great though!

Comment (Tastebuder) [reply src/test/resources/AnalyzeResponse.json:1]: Done 👍🏼

**Review by CameronMukherjee — APPROVED** [review]
LGTM

**Pull request merged**